src/NLog/Internal/StringHelpers.cs

```diff
@@ -59,5 +61,72 @@ internal static bool IsNullOrWhiteSpace(string value)
             return string.IsNullOrWhiteSpace(value);
 #endif
         }
+
+        /// <summary>
+        /// Replace string with <paramref name="comparison"/>
+        /// </summary>
+        /// <param name="str"></param>
+        /// <param name="oldValue"></param>
+        /// <param name="newValue"></param>
+        /// <param name="comparison"></param>
+        /// <returns>The same reference of nothing has been replaced.</returns>
+        public static string Replace([NotNull] this string str, [NotNull] string oldValue, string newValue, StringComparison comparison)
+        {
+            if (str == null)
+            {
+                throw new ArgumentNullException(nameof(str));
+            }
+
+            if (oldValue == null)
+            {
+                throw new ArgumentNullException(nameof(oldValue));
+            }
+
+            if (str.Length == 0)
+            {
+                //nothing todo
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1135)